assets/js/state/sagas/hosts.test.js

```diff
@@ -1,9 +1,90 @@
+import { runSaga } from 'redux-saga';
 import { recordSaga } from '@lib/test-utils';
-import { hostDeregistered } from '@state/sagas/hosts';
-import { removeHost } from '@state/hosts';
+import {
+  markDeregisterableHosts,
+  hostDeregisterable,
+  matchHost,
+  checkHostDeregisterable,
+  hostDeregistered,
+} from '@state/sagas/hosts';
+import {
+  cancelCheckHostIsDeregisterable,
+  setHostsDeregisterable,
+  removeHost,
+} from '@state/hosts';
 import { hostFactory } from '@lib/test-utils/factories';
 
 describe('Hosts sagas', () => {
+  it('should mark hosts as deregisterable', async () => {
+    const passingHost = hostFactory.build({ heartbeat: 'passing' });
+    const criticalHost = hostFactory.build({ heartbeat: 'critical' });
+    const unknownHost = hostFactory.build({ heartbeat: 'unknown' });
+
+    const dispatched = await recordSaga(markDeregisterableHosts, [
+      passingHost,
+      criticalHost,
+      unknownHost,
+    ]);
+
+    expect(dispatched).toContainEqual(
+      setHostsDeregisterable([criticalHost, unknownHost])
+    );
+  });
+
+  it('should mark a host as deregisterable', async () => {
+    const host = hostFactory.build();
+
+    const dispatched = [];
+
+    await runSaga(
```

Comment:
I've had to use `redux-saga`'s `runSaga()` function in this file because our `recordSaga()` utility function does not allow for multiple parameters to be passed to functions; which is required in order to test `hostDeregisterable()` and `checkHostDeregisterable()`. Thoughts on this?